src/ppp_device.c

```diff
@@ -395,46 +365,37 @@ static rt_err_t ppp_device_init(struct rt_device *device)
 static rt_err_t ppp_device_open(struct rt_device *device, rt_uint16_t oflag)
 {
     int result = RT_EOK;
+    rt_uint16_t uart_oflag;
     RT_ASSERT(device != RT_NULL);
 
     struct ppp_device *ppp_device = (struct ppp_device *)device;
-    static rt_device_t serial = RT_NULL;
 
-    ppp_device->ppp_link_status = RT_TRUE;
-    /* Creat a thread to creat ppp recieve function */
-    result = ppp_recv_entry_creat(ppp_device);
-    if (result != RT_EOK)
+    uart_oflag = RT_DEVICE_OFLAG_RDWR;
+    if (ppp_device->uart->flag & RT_DEVICE_FLAG_DMA_RX)
+        uart_oflag |= RT_DEVICE_FLAG_DMA_RX;
+    else if (ppp_device->uart->flag & RT_DEVICE_FLAG_INT_RX)
+        uart_oflag = RT_DEVICE_FLAG_INT_RX;
+    if (ppp_device->uart->flag & RT_DEVICE_FLAG_DMA_TX)
+        uart_oflag |= RT_DEVICE_FLAG_DMA_TX;
+    else if (ppp_device->uart->flag & RT_DEVICE_FLAG_INT_TX)
+        uart_oflag |= RT_DEVICE_FLAG_INT_TX;
+
+    if (rt_device_open(ppp_device->uart, uart_oflag))
```

Comment:
```suggestion
    if (rt_device_open(ppp_device->uart, uart_oflag) != RT_EOK)
```
返回值类型是 rt_err_t ，这样判断会更直观点